src/maxminddb.c

```diff
@@ -1116,13 +1119,13 @@ int MMDB_get_value(MMDB_entry_s *const start,
 int MMDB_vget_value(MMDB_entry_s *const start,
                     MMDB_entry_data_s *const entry_data,
                     va_list va_path) {
-    int length = path_length(va_path);
+    size_t length = path_length(va_path);
     const char *path_elem;
     int i = 0;
 
-    MAYBE_CHECK_SIZE_OVERFLOW(length,
-                              SIZE_MAX / sizeof(const char *) - 1,
-                              MMDB_INVALID_METADATA_ERROR);
+    if (length == SIZE_MAX) {
+        return MMDB_OUT_OF_MEMORY_ERROR;
```

Comment:
Is the new `MMDB_OUT_OF_MEMORY_ERROR` error more appropriate than the original `MMDB_INVALID_METADATA_ERROR` error?

Reply:
Good point!